### ISSUE TITLE
environment variable NETCDF4_DIR ignored

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sjaaky74/netcdf4-python#51]
```
What steps will reproduce the problem?
1. SEE BELOW FOR TRANSCRIPT
2.
3.

What is the expected output? What do you see instead?


What version of the product are you using? On what operating system?

ubuntu 10.04: have packages
  libnetcdf4
  libnetcdf-dev
  libhdf5-openmpi-1.8.4
  libhdf5-openmpi-dev

Please provide any additional information below.

bueler@bueler-pogo:~/usr/local/netCDF4-0.9.1$ locate libnetcdf.so.4
/usr/lib/libnetcdf.so.4
/usr/lib/libnetcdf.so.4.0.0
bueler@bueler-pogo:~/usr/local/netCDF4-0.9.1$ export NETCDF4_DIR=/usr/lib
bueler@bueler-pogo:~/usr/local/netCDF4-0.9.1$ echo $NETCDF4_DIR
/usr/lib
bueler@bueler-pogo:~/usr/local/netCDF4-0.9.1$ sudo python setup.py install

HDF5_DIR environment variable not set, checking some standard locations ..,
checking /home/bueler ...
checking /usr/local ...
checking /sw ...
checking /opt ...
checking /opt/local ...
checking /usr ...
HDF5 found in /usr

NETCDF4_DIR environment variable not set, checking some standard locations ..,
checking /home/bueler ...
checking /usr/local ...
checking /sw ...
checking /opt ...
checking /opt/local ...
checking /usr ...
Traceback (most recent call last):
  File "setup.py", line 68, in <module>
    raise ValueError('did not find netCDF version 4 headers and libs')
ValueError: did not find netCDF version 4 headers and libs


```

Original issue reported on code.google.com by `elbueler...@gtempaccount.com` on 20 May 2010 at 7:48

**Comment by GoogleCodeExporter** [comment]
```
Environment variables are not passed to sudo by default.  You need to do

sudo env=NETCDF4_DIR python setup.py install
```

Original comment by `whitaker.jeffrey@gmail.com` on 20 May 2010 at 7:54

**Comment by GoogleCodeExporter** [comment]
```
That didn't work either.

These instructions in INSTALL are pretty much nonsensical to debian users:

"""
make sure numpy (at least version 1.2.1) is installed.
make sure HDF5 (at least version 1.8.4) and netcdf-4 (at least version 4.1)
are installed.   
'python setup.py install' installs the netCDF4 and netcdftime modules.
The environment variables HDF5_DIR and NETCDF4_DIR can be 
set to point to the locations of the HDF5 and netcdf-4 libraries and
headers before setup.py is run. If they are not, some standard locations
will be searched.
"""

Apparently what you mean is something like this:

"""
Install the libnetcdf4 package which is really netcdf 3.6.3 and does not depend 
on
HDF5 and is not netcdf-4.  Then do
  $ sudo bash
  $ export NETCDF3_DIR=/usr
  $ python setup-nc3.py install
"""

Odd that /usr is not tried as one of the possibilities.

Basically debian users needing the netCDF3 python module have to use libnetcdf4 
and
netcdf4-python, and then ignor most of the install instructions.  Very odd.




```

Original comment by `elbueler...@gtempaccount.com` on 20 May 2010 at 8:28

**Comment by GoogleCodeExporter** [comment]
```
The netCDF3 module is just a hack for those that can't (or won't) install new 
versions of netcdf-4/hdf5.  the 
netCDF4 module works fine for  netcdf3 formatted files.  You should be able to 
just do

sudo bash
export NETCDF4_DIR=<path to netCDF 4 lib>
export HDF5_DIR=<path to hdf5 dir>
python setup.py install

which is pretty much exactly what the install instructions say.


```

Original comment by `whitaker.jeffrey@gmail.com` on 20 May 2010 at 9:19

**Comment by GoogleCodeExporter** [comment]
Original comment by `whitaker.jeffrey@gmail.com` on 26 Feb 2014 at 2:04
- Changed state: **Fixed**
